core/src/main/java/com/google/errorprone/bugpatterns/UnnecessaryBoxedVariable.java

```diff
@@ -287,7 +287,8 @@ private static boolean isBoxableType(Type type, VisitorState state) {
 
   private static boolean canChangeMethodSignature(VisitorState state, MethodSymbol methodSymbol) {
     return !ASTHelpers.methodCanBeOverridden(methodSymbol)
-        && ASTHelpers.findSuperMethods(methodSymbol, state.getTypes()).isEmpty();
+        && ASTHelpers.findSuperMethods(methodSymbol, state.getTypes()).isEmpty()
+        && !ASTHelpers.isRecord(methodSymbol);
```

Comment:
Performance-wise it seems better to test `!ASTHelpers.isRecord(methodSymbol)` first; its a cheaper test than the other two (especially `findSuperMethods`).